idaes/tests/test_import.py

```diff
@@ -11,81 +11,285 @@
+        else:
+            # Parse output of '-X importtime'
+            prev_import = None
+            for line in output.splitlines():
+                # For rows with timings
+                if line.startswith("import time:"):
+                    # Save if match with current module
+                    if line.endswith(module):
+                        times = re.findall(r"\d+", line)
+                        result.timing = float(times[1]) / 1e6
+                    # In any case, remember module name
+                    prev_import = line[line.rfind("|") + 1:].strip()
+                elif not allow_output:
+                    # Associate dis-allowed output with last seen module name
+                    result.add_error(f"[{prev_import}] {line}")
+
+    # Return timing and/or error information
+    return result
+
+
+####################################################################################
 
 
 @pytest.mark.unit
-def test_import():
-    root_dir = pathlib.Path(idaes.__file__).parent
-    failures, total = importr(root_dir)
-    n = len(failures)
-    assert n == 0, f"{n:d} failures in {total:d} tests: {failures}"
+@pytest.mark.parametrize("module_path", list(idaes_path.rglob("*.py")))
+def test_toplevel_statements(module_path):
+    """Look for "too many" top-level statements.
+
+    This test is far quicker than `test_import`, but they test different things.
+    It does not have any way of knowing how long it takes to import the module
+    (or whether it can be imported).
+
+    It also has a pretty crude idea of what a top-level statement is (and isn't)
+    """
+    # Skip some paths
+    if module_path == idaes_path / "__init__.py":
+        return
+
+    with module_path.open(encoding="utf8") as f:
+        num_statements = count_statements(f)
+        assert num_statements <= TOPLEVEL_LINES_MAX
+
+
+def count_statements(f):
+    statements = 0
+    comment_block = False
+    lineno = 0
+
+    # Look at file line-by-line
+    for line in f:
+        lineno += 1
+        is_statement = False
+        # if in a comment block, wait for it to end
+        if comment_block:
+            if re.match(r".*('''|\"\"\")", line):
+                comment_block = False
+        # look for start of comment block
+        elif re.match(r"\s*('''|\"\"\")", line):
+            comment_block = True
+        # skip empy, indented, comment lines
+        elif len(line) == 0 or re.match(r"^\s+$", line) or re.match(r"\s|#", line):
+            pass
+        # skip function, class defs or decorators
+        elif line.startswith("def") or line.startswith("class") or line.startswith("@"):
+            pass
+        # match anything that looks like a function call, except some special ones
+        elif re.match(r".*\w+\(", line):
+            is_statement = True
+            # these things *are* allowed at the top-level
+            if re.match(
+                ".*(add_command|CONFIG|list\(|dict\(|attempt_import|\.declare\()", line
```

Comment:
The RE is missing the raw string indicator (`r"`)

You should also probably skip "import".

You also might consider "guarding" these patterns with `\b`:
```python
if re.match(
    r".*\b(add_command|CONFIG|list|dict|attempt_import|declare)\b", line
    ):
```

Reply:
import is skipped because it doesn't look like a function. but ok on the other stuff.